File: prozesse_und_threads/loesung/05_orphan_process_repair.py
#!/usr/bin/env python3

import sys
import os
import time

if __name__ == "__main__":
    limit = 100
    pid = os.fork()

    if pid < 0:
        print("Fork failed")
        sys.exit(-1)
    elif pid == 0:
        print("begin child")
        time.sleep(3)
        print("PID:", os.getpid(), "PPID:", os.getppid(), "(child)")
        time.sleep(3)
        print("end child")
        os._exit(0)
    else:
        print("begin parent")
        print("PID:", os.getpid(), "PPID:", os.getppid(), "(parent)")
        print("Waiting for child process ...")
        os.wait()
        print("end parent")
        sys.exit(0)
    

File: prozesse_und_threads/loesung/05_orphan_process_repair.cpp
#include <unistd.h>
#include <iostream>
#include <sys/wait.h>

using namespace std;

int main()
{
    pid_t pid = fork();

    if (pid == 0)
    {
        cout << "begin child" << endl;
        sleep(3);
        cout << "PID: " << getpid() << " PPID: " << getppid() << " (parent)" << endl;
        sleep(3);
        cout << "end child" << endl;
    }
    else if (pid > 0)
    {
        int status;
        cout << "PID: " << getpid() << " PPID: " << getppid() << " (child)" << endl;
        cout << "Not waiting for my child!" << endl;
        wait(&status);
    }
    else
    {
        cout << "Fork failed!" << endl;
        return -1;
    }
    return 0;
}